cmd/maya-apiserver/app/server/snapshot_endpoint.go

```diff
@@ -167,6 +169,17 @@ func (s *HTTPServer) snapshotList(resp http.ResponseWriter, req *http.Request, v
 
 }
 
+// SnapshotClone is http handler for restore a snapshot to a persistent volume
+func (s *HTTPServer) snapshotClone(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	clone, err := s.volumeAdd(resp, req)
```

Comment:
It is adding a volume. Why are we calling it as a clone.

Reply:
Let us not reuse the methods here & inject `if else` conditionals throughout the code.
We will be better off if we go for well defined/dedicated structures that handle clone & snapshot respectively.